src/main/resources/assets/js/components/AppComponent.jsx

```diff
@@ -4,8 +4,16 @@ var React = require("react/addons");
 var AppHealthComponent = require("../components/AppHealthComponent");
 
 var AppComponent = React.createClass({
+  propTypes: {
+    model: React.PropTypes.object.isRequired,
+    router: React.PropTypes.object.isRequired
+  },
+
   onClick: function () {
-    this.props.onClick(this.props.model);
+    this.props.router.navigate(
```

Comment:
This is functionality that the [Flux](https://facebook.github.io/flux/docs/overview.html) design pattern is designed to help: rather than have deeply-nested descendant components know about a top-level app state like the router, let descendants fire events to a global dispatcher and let the dispatcher know about top-level app state.

Still, great work.